File: tmp_refactor.py
import re

path = 'c:/Users/User-3/Projects/BTGS/TGS FRONTEND/src/components/trips/DynamicExpenseGrid.jsx'

with open(path, 'r', encoding='utf-8') as f:
    text = f.read()

def replacer(match):
    full = match.group(0)
    func = match.group(2)
    field = match.group(3)
    if 'onBlur' in full:
        return full
    return full + f" onBlur={{e => validateFieldInline(row.id, '{field}', e.target.value)}}"

pattern = r"(onChange=\{e => (updateDetails|updateRow|updateTimeDetails)\(row\.id,\s*'(\w+)',\s*e\.target\.value\)\})"

new_text, count = re.subn(pattern, replacer, text)

print(f'Replaced {count} instances.')

with open(path, 'w', encoding='utf-8') as f:
    f.write(new_text)

File: tmp_refactor.js
const fs = require('fs');
const path = require('path');
const dir = 'c:/Users/User-3/Projects/BTGS/TGS FRONTEND/src/pages/';

let count = 0;
fs.readdirSync(dir).forEach(file => {
    if (file.endsWith('.jsx')) {
        let filepath = path.join(dir, file);
        let content = fs.readFileSync(filepath, 'utf8');

        // Target: <h1>Title</h1> \n <p>Subtitle description...</p>
        let regex = /(<h1>[\s\S]*?<\/h1>)\s*<p>[^<]*<\/p>/g;
        let newContent = content.replace(regex, '$1');

        if (content !== newContent) {
            fs.writeFileSync(filepath, newContent);
            console.log('Updated', file);
            count++;
        }
    }
});
console.log(`Finished checking. Updated ${count} files.`);
